control/mpc_lateral_controller/src/mpc.cpp

```diff
@@ -73,6 +73,20 @@ bool MPC::calculateMPC(
     return fail_warn_throttle("trajectory resampling failed. Stop MPC.");
   }
 
+  // get the diagnostic data
+  const auto [success_data_for_diagnostic, mpc_data_for_diagnostic] =
+    getData(mpc_resampled_ref_trajectory, current_steer, current_kinematics);
+  if (!success_data_for_diagnostic) {
+    return fail_warn_throttle("fail to get MPC Data for the diagnostic. Stop MPC.");
+  }
+
+  // get the diagnostic data w.r.t. the original trajectory
+  const auto [success_data_traj_raw, mpc_data_traj_raw] =
+    getData(mpc_traj_raw, current_steer, current_kinematics);
+  if (!success_data_traj_raw) {
+    return fail_warn_throttle("fail to get MPC Data for the raw trajectory. Stop MPC.");
+  }
```

Comment:
I do not agree with this part. Even if the calculation of the data for the diagnostics, since it does not affect the output of MPC, we should not stop MPC.
Like this, we have to be careful about the behavior of the node itself.

Reply:
fixed in d0cfc56e7a210e2590ea5c8864201cc5b0523be8 and dc98701f46db07d0a39892548285848d05f160e1